File: sample/src/main/java/io/github/wulkanowy/materialchipsinput/sample/MainActivity.java
package io.github.wulkanowy.materialchipsinput.sample;

import android.content.Intent;
import android.os.Bundle;
import android.widget.Button;

import androidx.appcompat.app.AppCompatActivity;
import androidx.fragment.app.Fragment;
import androidx.fragment.app.FragmentTransaction;

public class MainActivity extends AppCompatActivity {

    Button mContactListButton;
    Button mCustomChipsButton;
    private int mStackLevel = 0;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        mContactListButton.setOnClickListener(view -> {
            startActivity(new Intent(MainActivity.this, ContactListActivity.class));
        });
    }

    public void showDialog() {
        mStackLevel++;

        // DialogFragment.show() will take care of adding the fragment
        // in a transaction.  We also want to remove any currently showing
        // dialog, so make our own transaction and take care of that here.
        FragmentTransaction ft = getSupportFragmentManager().beginTransaction();
        Fragment prev = getSupportFragmentManager().findFragmentByTag("dialog");
        if (prev != null) {
            ft.remove(prev);
        }
        ft.addToBackStack(null);

        // Create and show the dialog.
        MyDialogFragment newFragment = MyDialogFragment.newInstance(mStackLevel);
        newFragment.show(ft, "ok");
    }
}

File: sample/src/main/java/io/github/wulkanowy/materialchipsinput/sample/MainActivity.kt
package io.github.wulkanowy.materialchipsinput.sample

import android.os.Bundle
import androidx.appcompat.app.AppCompatActivity
import io.github.wulkanowy.materialchipsinput.ChipItem
import io.github.wulkanowy.materialchipsinput.sample.databinding.ActivityMainBinding
import kotlin.random.Random

class MainActivity : AppCompatActivity() {

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        val binding = ActivityMainBinding.inflate(layoutInflater)
        setContentView(binding.root)

        val randomNumbers = mutableListOf<ChipItem>()

        repeat(30) {
            randomNumbers.add(
                SampleChipItem(Random.nextInt().toString(), Random.nextInt().toString())
            )
        }

        with(binding.mainChipsInput) {
            filterableChipItems = randomNumbers
            onTextChangeListener = {
                binding.scrollContainer.scrollTo(0, binding.scrollContainer.bottom)
            }
        }
    }
}

data class SampleChipItem(override val title: String, override val summary: String) : ChipItem

